### ISSUE TITLE
When trying top open remote file, VS Code opens new windows repeatedly until you kill it

### ISSUE THREAD

**Issue by j-h-a** [microsoft/vscode-remote-release#6249]
- VSCode Version (local): 1.63.2 899d46d82c4c95423fb7e10e68eba52050e30ba3 arm64
- VSCode Version: 1.63.2 899d46d82c4c95423fb7e10e68eba52050e30ba3 x64
- Local OS Version: macOS 12.1 (arm64 M1 Mac)
- Remote OS Version: Ubuntu 20.04.1
- Remote Extension/Connection Type: SSH

Steps to Reproduce:

The set-up:
1. Set up a remote SSH connection with your key.pub copied to the remote using `ssh-copy-id`, and your local `.ssh/config` set up to use that key for your remote host, such that you can log in without entering any passwords.
2. Use the command palette to open an SSH connection to the remote machine, and install `code` there.

Reproduce-the-issue:
3. Open a fresh VS Code locally.
4. Use command palette to connect to your remote host
5. Use command palette to create a new terminal on the remote host
6. In the remote terminal, cd into `~/.ssh` and type `code config`

Does this issue occur when you try this locally?: No

Expected behaviour:
Remote file opens in local code editor.

Actual behaviour:
VS Code opens a new window and establishes another SSH connection to the same remote in that window. As soon as that window has connected it performs the same task, opening another window. This repeats opening new VS Code windows until you kill VS Code entirely or run out of memory.

Additional notes:
I don't know if this matters, but...
* I am using ed25519 keys
* The remote `config` file already existed - I had used `scp` to put an example there before connecting to edit it.
* This happened during setup of a new remote, the remote was a fresh ubuntu 20.04 install.
* I was able to repeat this a few times, killing code and reopening and connecting to the remote again, but at some point it stopped doing it and instead displayed: Unable to connect to VS Code server. Error in request.

**Comment by roblourens** [comment]
So you want to open a window of the desktop client after connecting with Remote-SSH? We add a different `code` to your terminal's PATH which points at the server and lets you open files and folders through the Remote-SSH connection. To get what you want, you probably have to use the absolute path to your vscode install.

> As soon as that window has connected it performs the same task, opening another window.

I don't understand this part. It's opening multiple windows on the same file? What's in the new window? Maybe a screenshot would help.

**Comment by j-h-a** [comment]
> So you want to open a window of the desktop client after connecting with Remote-SSH?

No, I already have a Code window open, and it is connected to remote via ssh.

When I type `code README.md` in the terminal (in VS Code, connected to remote) what I want to happen is `README.md` should open in my existing Code window, i.e. in the editor pane above the terminal pane at the bottom where I just typed.

> I don't understand this part.

This is my description of the bug/issue.

Instead of opening the file, VS Code opens a new window with the same profile (like open recent) and it automatically connects to the same remote. The remote code-server is telling new windows to "open a new window" so this new window opens another new window and connects to the remote. VS Code gets stuck in a loop opening new windows and never opening the file.

**Comment by roblourens** [comment]
Any difference if you add a `-r` ("reuse window") to the code command?

A screenshot might help just to make sure I am following your description
